online/apply_functions/bbci_apply_sendControl.m

```diff
@@ -16,72 +16,80 @@
+        otherwise,
+            error('feedback.receiver unknown');
     end
-    pnet('closeall');
-    data_feedback.opt.socke= pnet('udpsocket', 1111); 
-    pnet(data_feedback.opt.socke, 'udpconnect', 'localhost', 9100');
-   case 'pyff',
-%    send_udp_xml('init', bbci_feedback.host, bbci_feedback.port);
-	pyff_sendUdp('init',  bbci_feedback.host, bbci_feedback.port);
-   case 'tobi_c',
-    send_tobi_c_udp('init', bbci_feedback.host, bbci_feedback.port);
-    case 'matlab',
-    if isfield(bbci_feedback, 'opt'),
-      data_feedback.opt= bbci_feedback.opt;
-    else
-      data_feedback.opt= struct;
-    end
-   otherwise,
-    error('feedback.receiver unknown');
-  end
-  return;
+    return;
 end
 
 %% - CLOSE
 if isequal(control_signal, 'close'),
-  switch(bbci_feedback.receiver),
-   case 'udp',
-		 pnet('closeall');
-%    if ~isempty(data_feedback.opt.socke),  
-%      pnet(data_feedback.opt.socke, 'close');
-%      data_feedback.opt.socke= [];
-%    end
-   case 'pyff',
-%    We do not close this channel, as it is probably still required to
-%    send signals to Pyff.
-%    send_udp_xml('close');
-   case 'tobi_c',
-    send_tobi_c_udp('close');
-  end
-  return;
+    switch(bbci_feedback.receiver),
+        case 'udp',
+            pnet('closeall');
+            %    if ~isempty(data_feedback.opt.socke),
+            %      pnet(data_feedback.opt.socke, 'close');
+            %      data_feedback.opt.socke= [];
+            %    end
+        case 'pyff',
+            %    We do not close this channel, as it is probably still required to
+            %    send signals to Pyff.
+            %    send_udp_xml('close');
+        case 'tobi_c',
+            send_tobi_c_udp('close');
+    end
+    return;
 end
 
 %% - Operation
 switch(bbci_feedback.receiver),
- case '',
-  % do nothing
- case 'udp',
-  if ~isempty(control_signal),
-    pnet(data_feedback.opt.socke, 'write', control_signal{:}); 
-    pnet(data_feedback.opt.socke, 'writepacket');
-	end
- case 'pyff',
-   if ~isempty(control_signal),
-     pyff_sendUdp(control_signal{:});
-   end
- case 'tobi_c',
-  if ~isempty(control_signal),
-    send_tobi_c_udp('send', control_signal{2}(1));
-  end
- case 'matlab',
-  idx= find(cellfun(@(x)isequal(x,'cl_output'), control_signal));
-  cls_output= control_signal{idx(end)+1};
-  data_feedback= bbci_feedback.fcn(cls_output, data_feedback);
+    case '',
+        % do nothing
+    case 'udp',
+        if ~isempty(control_signal),
+            pnet(data_feedback.opt.socke, 'write', control_signal{:});
+            pnet(data_feedback.opt.socke, 'writepacket');
+        end
+    case 'pyff',
+        if ~isempty(control_signal),
+            pyff_sendUdp(control_signal{:});
+        end
+    case 'tobi_c',
+        if ~isempty(control_signal),
+            send_tobi_c_udp('send', control_signal{2}(1));
+        end
+    case 'lsl',
+        if ~isempty(control_signal),
+            data_feedback.opt.outlet.push_sample(control_signal{2}(1));
```

Comment:
control_signal is only a key value pair in the pyff convention. For the feedbacks programmed in processing, it is a cell containing a double. I think that's the better way here. It makes no sense to create a key value pair, if the key is just discarded.